src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerOptions.cs

```diff
@@ -100,5 +102,146 @@ public int MaxQueueLength
             }
         }
+
+        internal void Configure(IConfiguration configuration)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (ParseBool(configuration, nameof(DisableColors), out bool disableColors))
+            {
+                DisableColors = disableColors;
+            }
+#pragma warning restore CS0618 // Type or member is obsolete
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (ParseEnum(configuration, nameof(Format), out ConsoleLoggerFormat format))
+            {
+                Format = format;
+            }
+#pragma warning restore CS0618 // Type or member is obsolete
+
+            if (configuration[nameof(FormatterName)] is string formatterName)
+            {
+                FormatterName = formatterName;
+            }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (ParseBool(configuration, nameof(IncludeScopes), out bool includeScopes))
+            {
+                IncludeScopes = includeScopes;
+            }
+#pragma warning restore CS0618 // Type or member is obsolete
+
+            if (ParseEnum(configuration, nameof(LogToStandardErrorThreshold), out LogLevel logToStandardErrorThreshold))
+            {
+                LogToStandardErrorThreshold = logToStandardErrorThreshold;
+            }
+
+            if (ParseInt(configuration, nameof(MaxQueueLength), out int maxQueueLength))
+            {
+                MaxQueueLength = maxQueueLength;
+            }
+
+            if (ParseEnum(configuration, nameof(QueueFullMode), out ConsoleLoggerQueueFullMode queueFullMode))
+            {
+                QueueFullMode = queueFullMode;
+            }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (configuration[nameof(TimestampFormat)] is string timestampFormat)
+            {
+                TimestampFormat = timestampFormat;
+            }
+#pragma warning restore CS0618 // Type or member is obsolete
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (ParseBool(configuration, nameof(UseUtcTimestamp), out bool useUtcTimestamp))
+            {
+                UseUtcTimestamp = useUtcTimestamp;
+            }
+#pragma warning restore CS0618 // Type or member is obsolete
+        }
+
+        /// <summary>
+        /// Parses the configuration value at the specified key into a bool.
+        /// </summary>
+        /// <returns>true if the value was successfully found and parsed. false if the key wasn't found.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when invalid data was found at the specified configuration key.</exception>
+        internal static bool ParseBool(IConfiguration configuration, string key, out bool value)
+        {
+            if (configuration[key] is string valueString)
+            {
+                try
+                {
+                    value = bool.Parse(valueString);
```

Comment:
Should this use `TryParse` instead?
TryParse is used internally too, also avoids catching and re-throwing the exception.

Similar at other places here.

Reply:
It is intentionally using Parse in order to get the exception + message to pass into the `InvalidOperationException`, which adds the "config path" to the exception information. This is following the same logic/behavior as [ConfigurationBinder](https://github.com/dotnet/runtime/blob/75eac9643f3d8952df3cffd928e7987887bb7dab/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L858-L865), which this code was refactored out of using.

If we used `TryParse`, we would need to come up with our own message to say why the value wasn't able to be parsed.

Lastly, we don't expect invalid data in the config. So the exception is only thrown in exceptional cases - not in "normal" control flows.